primitives/runtime/src/chain.rs

```diff
@@ -192,12 +192,46 @@ pub trait Chain: Send + Sync + 'static {
 	fn max_extrinsic_weight() -> Weight;
 }
 
+pub trait ChainShadow {
```

Comment:
@serban300 
cool, and now we can merge/replace `ChainShadow` with `UnderlyingChainProvider` from runtime-common - it is exactly the same trait, so I would just move UnderlyingChainProvider here instead of ChainShadow

```
/// A trait that provides the type of the underlying chain.
pub trait UnderlyingChainProvider {
	/// Underlying chain type.
	type Chain: Chain;
}
```

Reply:
and also this:
```
/// Underlying chain type.
pub type UnderlyingChainOf<C> = <C as UnderlyingChainProvider>::Chain;
```

can be moved to the primites here also